### PR TITLE
Encryption

### PR DESCRIPTION
**Pull request by cclevinger**

### CODE REVIEW

**Review by pdx-cs-lbryan — CHANGES_REQUESTED** [review]
Logging into dashboard doesn't work
Reviews don't work
Edit: On further testing by manually adding an unencrypted name for log in purposes, I was able to log into the dashboard
from there:
clicking a student's name to view reports doesn't work
editing functionality results in bugs
switching sessions resulted in a bug
the report list doesn't work

Please remember to follow the guide in the coding styles document. You should only be requesting a review after testing out the application and ensuring that currently implemented features have not been broken by your update and that your new features are functional. Bugs will happen, but as many of them as possible should be removed prior to asking for review.

File: gbmodel/model_sqlalchemy.py [hunk 1]
```diff
@@ -3,6 +3,23 @@
 import datetime
 from extensions import db
 from sqlalchemy import exc, func
+from cryptography.fernet import Fernet
+key_file = open("../key.txt")
```
Comment (pdx-cs-lbryan) [inline gbmodel/model_sqlalchemy.py]: should have a try/except here

File: gbmodel/model_sqlalchemy.py [hunk 2]
```diff
@@ -3,6 +3,23 @@
...
+key = key_file.readline()
+key = bytes(key.encode("UTF8"))
+cipher = Fernet(key)
+
+
+def encrypt(p_text):
```
Comment (pdx-cs-lbryan) [inline gbmodel/model_sqlalchemy.py]: needs comments, error handling

File: gbmodel/model_sqlalchemy.py [hunk 3]
```diff
@@ -3,6 +3,23 @@
...
+    c_text = cipher.encrypt(bytes(p_text, encoding='UTF8'))  # Encrypt name
+    return c_text
+
+
+def decrypt(c_text):
```
Comment (pdx-cs-lbryan) [inline gbmodel/model_sqlalchemy.py]: needs comments/error handling

File: gbmodel/model_sqlalchemy.py [hunk 4]
```diff
@@ -293,10 +310,16 @@ def check_dup_student(self, id, session_id):
         Output: return False if the student was already in
                 return True otherwise
         """
+        result = None
         try:
-            result = students.query.filter_by(id=id, session_id=session_id).first()
+            all_students = students.query.filter_by(session_id=session_id).all()
+            print(all_students)
```
Comment (pdx-cs-lbryan) [inline gbmodel/model_sqlalchemy.py]: not sure we want to be printing this

**Review by pdx-cs-lbryan — CHANGES_REQUESTED** [review]
key.txt should be removed
Some testing information would be very helpful

left some comments in the code.

doesn't look like professor info isn't encrypted, but I'm not sure what should be done there

it looks like review/reports information isn't being encrypted either

see slack comment

File: gbmodel/model_sqlalchemy.py (hunk 1 above)
Comment (pdx-cs-lbryan) [reply gbmodel/model_sqlalchemy.py]: this line is also giving me an error with the current file structure of the repo

**Comment by pdx-cs-lbryan** [comment]
A couple crashes I've found so far

When trying to remove students:
    st = students.query.filter(students.id == student.id,

    AttributeError: 'NoneType' object has no attribute 'id'

When trying to access the student dashboard:
    student_name = validate_student().name

    AttributeError: 'int' object has no attribute 'name'

**Comment by J8cob1** [comment]
Not needed anymore

**Comment by J8cob1** [comment]
Testing something...

**Pull request closed without merging**